modules/manually-restoring-cluster-etcd-backup.adoc

```diff
@@ -0,0 +1,312 @@
+  sed -E 's/.*\"(.*)\"/\1/g'
+----
+
+.. The value for `<ETCD_INITIAL_CLUSTER>` should be like the following, where `<ETCD_NAME_n>` is the `<ETCD_NAME>` of each control plane host. 
++
+[NOTE]
+====
+Note that the port used here is 2380 and not 2379.
+====
++
+.Sample output
+[source,terminal]
+----
+<ETCD_NAME_0>=https://<IP_HOST_0>:2380,<ETCD_NAME_1>=https://<IP_HOST_1>:2380,<ETCD_NAME_2>=https://<IP_HOST_2>:2380
+----
++
+This `<ETCD_INITIAL_CLUSTER>` value is the same for all control plane hosts.
+
+. Regenerate the etcd database from the backup
+
+.. Copy the `etcd` backup to `/var/lib/etcd`:
++
+[source,terminal]
+----
+$ sudo cp /home/core/assets/backup/<snapshot_yyyy-mm-dd_hhmmss>.db /var/lib/etcd
+----
+
+.. Start a new container which includes the etcdctl tool and use a plain bash entrypoint:
++
+[source,terminal]
+----
+$ sudo podman run --rm --authfile=/var/lib/kubelet/config.json -it --entrypoint="/bin/bash" -v /var/lib/etcd:/var/lib/etcd:z registry.redhat.io/openshift4/ose-etcd:latest
+----
+
+.. Check that the version of the `etcdctl` tool is >= the version of theetcd server where the backup was created:
++
+[source,terminal]
+----
+$ sudo etcdctl version
+----
+
+.. Run the command to regenerate the etcd database, using the correct values for the current host. 
++
+[source,terminal]
+----
+$ sudo ETCDCTL_API=3 /usr/bin/etcdctl snapshot restore /var/lib/etcd/<snapshot_yyyy-mm-dd_hhmmss>.db \
+  --name "<ETCD_NAME>" \
+  --initial-cluster="<ETCD_INITIAL_CLUSTER>" \
+  --initial-cluster-token "openshift-etcd-<UUID>" \
+  --initial-advertise-peer-urls "<ETCD_NODE_PEER_URL>" \
+  --data-dir="/var/lib/etcd/restore-<UUID>" \
+  --skip-hash-check=true
+----
++
+[NOTE]
+====
+Note the the quotes are mandatory here.
+====
+
+. Write down the values printed in the added member logs; for example:
++
+.Example output
+----
+2022-06-28T19:52:43Z    info    membership/cluster.go:421   added member    {"cluster-id": "c5996b7c11c30d6b", "local-member-id": "0", "added-peer-id": "56cd73b614699e7", "added-peer-peer-urls": ["https://10.0.91.5:2380"], "added-peer-is-learner": false}
+2022-06-28T19:52:43Z    info    membership/cluster.go:421   added member    {"cluster-id": "c5996b7c11c30d6b", "local-member-id": "0", "added-peer-id": "1f63d01b31bb9a9e", "added-peer-peer-urls": ["https://10.0.90.221:2380"], "added-peer-is-learner": false}
+2022-06-28T19:52:43Z    info    membership/cluster.go:421   added member    {"cluster-id": "c5996b7c11c30d6b", "local-member-id": "0", "added-peer-id": "fdc2725b3b70127c", "added-peer-peer-urls": ["https://10.0.94.214:2380"], "added-peer-is-learner": false}
+----
+.. Exit from the container.
+
+.. Repeat these steps on all the control plane hosts, checking that the values printed in the added member logs are the same.
+
+. Move the regenerated etcd database to the default location.
+
+.. Move the regenerated database (that is, the `member` folder) in the etcd default location `/var/lib/etcd`:
```

Comment:
```suggestion
.. Move the regenerated database (that is, the `member` folder created by the previous`etcdctl snapshot restore`) in the etcd default location `/var/lib/etcd`:
```

Reply:
Is it more clear this way ?

Reply:
Yes, this is actually better than before.